### PR TITLE
Update search-npm extension

### LINKED ISSUE #15790: [Search npm Packages] Git URLs aren't being parsed properly on "open repository" action
### Extension

https://www.raycast.com/mrmartineau/search-npm

### Raycast Version

1.87.4

### macOS Version

15.1.1

### Description

The default open action is "Open Repository". When attempting to open any package with this extension, I'm seeing the following finder error:

This was working previously, I only started noticing this issue in the last week or so.

### Steps To Reproduce

1. Make sure your "Default Open Action" is set to "Open Repository"

   

2. In Raycast, open up the "Search npm packages" command
3. Search for "express" 
4. Select the "express" package — observe the error:

https://github.com/user-attachments/assets/3f07bf5e-6222-4c75-b4fd-8649b8c261e0

### Current Behaviour

Git URLs from npm are not being parsed properly and finder throws an error

### Expected Behaviour

I would expect that git URLs would be parsed properly and that my web browser would open with the express repository.

### PR DESCRIPTION
**Pull request by ridemountainpig**
## Description
Added a parser for GitHub repository URL in the open repository action.
Closes #15790

## Screencast

https://github.com/user-attachments/assets/80186a43-8ab5-408b-8bb7-324512418ff8

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by LitoMore — CHANGES_REQUESTED** [review]

File: extensions/search-npm/src/components/PackagListItem.tsx [hunk 1]
```diff
@@ -72,11 +72,16 @@ export const PackageListItem = ({
     handleFaveChange?.()
   }
 
+  const extractGithubUrl = (repositoryUrl: string): string => {
+    const match = repositoryUrl.match(/github\.com\/[^\s]+/)
+    return match ? `https://${match[0].replace(/\.git$/, '')}` : repositoryUrl
+  }
```
[the remark this answers is not in the document]
Comment (ridemountainpig) [reply extensions/search-npm/src/components/PackagListItem.tsx]: > This only works with GitHub. Please consider using https://www.npmjs.com/package/git-url-parse instead for URL parsing.

I have updated the code to use Git URL parsing. Thank you.

File: extensions/search-npm/package.json [hunk 2]
```diff
@@ -132,7 +133,9 @@
   "dependencies": {
     "@raycast/api": "^1.83.2",
     "@raycast/utils": "^1.17.0",
+    "@types/git-url-parse": "^9.0.3",
```
Comment (LitoMore) [inline extensions/search-npm/package.json:136]: Let's move the DefinitelyTyped dependency to `devDependencies`.

**Review by LitoMore — CHANGES_REQUESTED** [review]

File: extensions/search-npm/package.json (hunk 2 above)
Comment (ridemountainpig) [reply extensions/search-npm/package.json]: Updated, thank you.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**